java/java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java

```diff
@@ -3008,10 +3027,194 @@ private void insideBreakOrContinue(Env env) throws IOException {
             }
         }
     }
+    
+    private void addClassTypes(final Env env, DeclaredType baseType) throws IOException{
+        EnumSet<ElementKind> classKinds = EnumSet.of(CLASS, INTERFACE, ENUM, ANNOTATION_TYPE, TYPE_PARAMETER);
+        if (isRecordSupported(env, null)) {
+            classKinds.add(TreeShims.getRecordKind());
+        }
+        addTypes(env, classKinds, null);
+    }
+
+    private boolean isRecordSupported(final Env env, String prefix) {
```

Comment:
I think this method's name and content do not really match - it is not only "isRecordSupported", but also "and is the prefix a prefix of 'record''. I would suggest to drop the second concern from this method. Many callers don't need the prefix check. I counted just two callers that really need to check the prefix (for many, the prefix check happens automatically later), and I think it would be cleaner for them to call the prefix check explicitly.

Also, for discussion - should there be a preview enablement test here as well?

Reply:
Corrected the method signature now